### PR TITLE
made-js feature/SOF-5214

### PR DESCRIPTION
**Pull request by adewyer**
lattice update from scalar to additive

### CODE REVIEW

File: src/basis/basis.js [hunk 1]
```diff
@@ -396,14 +397,23 @@ export class Basis {
...
             }
+        } else if(this._elements.array.length === 1) {
+            const element = this._elements.getArrayElementByIndex(0);
+            Object.keys(PERIODIC_TABLE).forEach(key => {
+                if (element === PERIODIC_TABLE[key].symbol) {
+                    maxDistance = math.precise(PERIODIC_TABLE[key].atomic_radius_pm * 0.01, 4);
+                }
+            });
         }
         if (maxDistance === 0) {
```
Comment (adewyer) [inline src/basis/basis.js:418]: We should be able to remove this if statement, no structures with `elements.array.length <= 0`

File: src/basis/basis.js [hunk 2]
```diff
@@ -396,14 +397,23 @@ export class Basis {
...
+                    }
                 }
             }
+        } else if(this._elements.array.length === 1) {
+            const element = this._elements.getArrayElementByIndex(0);
+            Object.keys(PERIODIC_TABLE).forEach(key => {
+                if (element === PERIODIC_TABLE[key].symbol) {
+                    maxDistance = math.precise(PERIODIC_TABLE[key].atomic_radius_pm * 0.01, 4);
+                }
+            });
```
Comment (timurbazhirov) [inline src/basis/basis.js:416]: (1) I suggest moving the lines 411-416 to a separate function on documenting exactly what's happening therein, especially the "magic" numbers of 0.01 and 4. This new function has to be tested also.

(2) For the case when we have an atom that does not exist in the periodic table (say, "Si1" the if condition won't ever be satisfied

(3) The Foreach is a very inefficient implementation, we could just use `find`

File: src/lattice/lattice.js [hunk 3]
```diff
@@ -10,6 +10,8 @@ import {LatticeVectors} from "./lattice_vectors";
 import {primitiveCell} from "../cell/primitive_cell";
 import {Cell} from "../cell/cell";
 
+export const nonPeriodicLatticeAdditiveValue = 2.0;
```
Comment (timurbazhirov) [inline src/lattice/lattice.js:13]: (1) we could also call this `nonPeriodicLatticeMinimumSizeAngstroms` 
(2) let's add a comment explaining what this is
(3) I think it should be used in https://github.com/Exabyte-io/exabyte-web-app/pull/1660/files# also

File: src/basis/basis.js [hunk 4]
```diff
@@ -396,18 +413,41 @@ export class Basis {
...
+                for (let j = i + 1; j < this._elements.array.length; j++) {
+                    const distance = math.vDist(
+                        this._coordinates.getArrayElementByIndex(i), this._coordinates.getArrayElementByIndex(j));
+                    if (distance > maxDistance) {
+                        maxDistance = distance;
+                    }
                 }
             }
         }
+        console.log(maxDistance);
```
Comment (adewyer) [inline src/basis/basis.js:427]: Will remove on next set of revisions.

**Comment by adewyer** [comment]
New basis tests passing locally.
Screen Shot 2021-12-03 at 1 13 37 PM
Screen Shot 2021-12-03 at 1 13 43 PM

File: src/basis/basis.js [hunk 5]
```diff
@@ -396,18 +413,41 @@ export class Basis {
...
+     * @returns {Number}
+    */
+    getElementAtomicRadius() {
+        const elementSymbol = this._elements.getArrayElementByIndex(0)
+        const elementName = Object.keys(PERIODIC_TABLE).find(key => elementSymbol === PERIODIC_TABLE[key].symbol);
+        if (elementName) {
+            return math.precise((PERIODIC_TABLE[elementName].atomic_radius_pm * 0.01), 4);
+        }
+        return 1.0;
+    }
```
Comment (timurbazhirov) [inline src/basis/basis.js:449]: Maybe we should move the above function to the Periodic Table package, it seems like a good place for it.

File: src/basis/basis.js [hunk 6]
```diff
@@ -378,6 +379,22 @@ export class Basis {
...
+    /**
+     * @summary function returns the minimum basis lattice size for a structure.
+     * The lattice size is based on the atomic radius of an element if the basis contains a single atom.
+     * The lattice size is based on the maximum pairwise distance across a structure if the basis contains > 2 atoms.
+     * @returns {Number}
+     */
+    minimumBasisLatticeSize() {
+        let singleAtomAtomicRadii = 0;
+        if (this._elements.array.length === 1) {
+            singleAtomAtomicRadii = this.getElementAtomicRadius();
```
Comment (timurbazhirov) [inline src/basis/basis.js:391]: I'd just call it `addiveContribution` or something like that

File: src/basis/basis.js [hunk 7]
```diff
@@ -1,12 +1,13 @@
 import _ from "underscore";
 import s from "underscore.string";
-import {getElectronegativity} from "@exabyte-io/periodic-table.js";
+import {PERIODIC_TABLE, getElectronegativity} from "@exabyte-io/periodic-table.js";
 
 import math from "../math";
-import {Lattice} from "../lattice/lattice";
+import {Lattice, nonPeriodicLatticeMinimumSizeAngstroms} from "../lattice/lattice";
```
Comment (timurbazhirov) [inline src/basis/basis.js:6]: I thought this has to be a factor?? The distance is calculated in `maxPairwiseDistance`

File: src/basis/basis.js [hunk 8]
```diff
@@ -378,6 +379,22 @@ export class Basis {
             .some(x => !x);
     }
 
+    /**
+     * @summary function returns the minimum basis lattice size for a structure.
+     * The lattice size is based on the atomic radius of an element if the basis contains a single atom.
+     * The lattice size is based on the maximum pairwise distance across a structure if the basis contains > 2 atoms.
+     * @returns {Number}
+     */
+    minimumBasisLatticeSize() {
```
Comment (timurbazhirov) [inline src/basis/basis.js:388]: `get minimumLatticeSize` ?

File: src/lattice/lattice.js [hunk 9]
```diff
@@ -10,6 +10,14 @@ import {LatticeVectors} from "./lattice_vectors";
 import {primitiveCell} from "../cell/primitive_cell";
 import {Cell} from "../cell/cell";
 
+/**
+ * Additive factor used to calculate the new lattice size for non-periodic systems.
+ * The ensures that a non-periodic structure will have a minimum lattice size of 2x2x2.
+ * The minimum lattice size ensures that the structures coordinates will fit inside the cell
+ * used to hold the structure.
+ */
+export const nonPeriodicLatticeMinimumSizeAngstroms = 2.0;
```
Comment (timurbazhirov) [inline src/lattice/lattice.js:19]: This is clearly not Angstrom?

File: src/basis/basis.js (hunk 4 above)
Comment (timurbazhirov) [reply src/basis/basis.js:427]: We might as well just clean it ahead of time or not commit at all ...

File: src/basis/basis.js (hunk 5 above)
Comment (adewyer) [reply src/basis/basis.js:449]: moved to periodic-table

File: package.json [hunk 10]
```diff
@@ -39,6 +39,6 @@
         "mixwith": "^0.1.1",
         "underscore": "^1.8.3",
         "underscore.string": "^3.3.4",
-        "@exabyte-io/periodic-table.js": "2.0.4"
+        "@exabyte-io/periodic-table.js": "file:/stack/lib/periodic-table"
```
Comment (adewyer) [inline package.json:42]: To run this code you need to link to periodic-table branch SOF-5214.

File: src/basis/basis.js [hunk 11]
```diff
@@ -378,6 +379,23 @@ export class Basis {
...
+     * @summary function returns the minimum basis lattice size for a structure.
+     * The lattice size is based on the atomic radius of an element if the basis contains a single atom.
+     * The lattice size is based on the maximum pairwise distance across a structure if the basis contains > 2 atoms.
+     * @returns {Number}
+     */
+    getMinimumBasisLatticeSize() {
+        let latticeSizeAdditiveContribution = 0;
+        if (this._elements.array.length === 1) {
+            const elementSymbol = this._elements.getArrayElementByIndex(0)
+            latticeSizeAdditiveContribution = getElementAtomicRadius(elementSymbol);
```
Comment (adewyer) [inline src/basis/basis.js:392]: `getElementAtomicRadius` function was moved to Periodic-Table - https://github.com/Exabyte-io/periodic-table.js/pull/11/files

File: src/basis/basis.js [hunk 12]
```diff
@@ -378,6 +378,23 @@ export class Basis {
             .some(x => !x);
     }
 
+    /**
+     * @summary function returns the minimum basis lattice size for a structure.
+     * The lattice size is based on the atomic radius of an element if the basis contains a single atom.
+     * The lattice size is based on the maximum pairwise distance across a structure if the basis contains > 2 atoms.
+     * @returns {Number}
+     */
+    getMinimumBasisLatticeSize() {
```
Comment (timurbazhirov) [inline src/basis/basis.js:387]: I'd just make it a getter and call `minLatticeSize`

```js
get minLatticeSize
```

File: package.json (hunk 10 above)
Comment (timurbazhirov) [reply package.json:42]: Then maybe we should point to that branch/hash instead inside when committing the code??

File: tests/basis/basis.js [hunk 13]
```diff
@@ -162,6 +162,22 @@ describe('Basis', function () {
...
+        const basis = new Basis(C2H4.basis);
+        const minimumBasisSize = 3.162;
+        const latticeSize = basis.getMinimumLatticeSize();
+        expect(latticeSize).to.be.equal(minimumBasisSize);
+    })
+
+    //** Minimum Lattice Size Atom//
+    it('should return minimum lattice size for an atom', function() {
+        const basis = new Basis(Na.basis);
+        const minimumBasisSize = 1.9;
```
Comment (timurbazhirov) [inline tests/basis/basis.js:176]: What is "minimumBasisSize" - is this a minimum pairwise distance for C2H4 and atomic radius for Na?? If yes, we should call them accordingly for clarity or add commentaries

File: src/material.js [hunk 14]
```diff
@@ -80,7 +80,8 @@ export class Material {
         return {
```
Comment (adewyer) [inline src/material.js:80]: Changes to this file are from merging SOF-5098 into this branch.

**Pull request merged**